File: go-gin/services/jwt_service/user_gen_jwt.go
package jwt_service

import (
	"go1/usecase"
)

type UserGenJwt struct {
	Username string
	Role     string
}

func FromLoginForm(loginForm usecase.LoginForm) *UserGenJwt {
	return &UserGenJwt{
		Username: loginForm.Username,
		Role:     "Admin",
	}
}

File: php-hyperf/app/Services/Jwt/UserGenJwt.php
<?php

declare(strict_types=1);


namespace App\Services\Jwt;

use App\Services\Login\LoginForm;

readonly class UserGenJwt
{
    private function __construct(
        public string $username,
        public string $role,
    )
    {
    }

    public static function createFromLoginForm(LoginForm $loginForm): UserGenJwt
    {
        return new self($loginForm->username, 'Admin');
    }
}